src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -864,10 +870,45 @@ private List<DataProviderMetadata> GetDataProvidersMetadata(DataProviders.DataPr
             return dataprovidersMetadata;
         }
 
-        private bool VerifyEntity(EntityInvoker invoker, ref QueryResponse<DiagnosticApiResponse> queryRes)
+        private bool VerifyEntity(EntityInvoker invoker, ref QueryResponse<DiagnosticApiResponse> queryRes, string publishingDetectorId)
         {
             List<EntityInvoker> allDetectors = this._invokerCache.GetAll().ToList();
 
+            var detectorWithSameId = allDetectors.FirstOrDefault(d => d.EntryPointDefinitionAttribute.Id == invoker.EntryPointDefinitionAttribute.Id);
```

Comment:
We probably want to make this case insensitive?

Reply:
Since at line 895 we are doing case insensitive comparisons

Reply:
Done!